core/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/models.scala

```diff
@@ -0,0 +1,146 @@
+package org.broadinstitute.dsde.workbench.leonardo
+
+import java.net.URL
+import java.time.Instant
+
+import ca.mrvisser.sealerate
+import cats.implicits._
+import enumeratum.{Enum, EnumEntry}
+import org.broadinstitute.dsde.workbench.model.WorkbenchEmail
+import org.broadinstitute.dsde.workbench.model.google.{GcsPath, parseGcsPath}
+
+import scala.collection.immutable
+import scala.util.matching.Regex
+
+
+final case class MachineType(value: String) extends AnyVal
+
+sealed trait CloudService extends EnumEntry with Product with Serializable {
+  def asString: String
+
+  override def toString: String = asString //Enumeratum's withName function uses `toString` as key for lookup
+}
+object CloudService extends Enum[CloudService]{
+  case object Dataproc extends CloudService {
+    val asString: String = "DATAPROC"
+  }
+  case object GCE extends CloudService {
+    val asString: String = "GCE"
+  }
+
+  override def values: immutable.IndexedSeq[CloudService] = findValues
+}
+
+final case class ClusterInternalId(asString: String) extends AnyVal
+
+sealed trait RuntimeConfig extends Product with Serializable {
+  def cloudService: CloudService
+  def machineType: MachineType
+  def diskSize: Int
+}
+object RuntimeConfig {
+  final case class GceConfig(
+                              machineType: MachineType,
+                              diskSize: Int
+                            ) extends RuntimeConfig {
+    val cloudService: CloudService = CloudService.GCE
+  }
+
+  final case class DataprocConfig(numberOfWorkers: Int,
+                                  masterMachineType: String,
+                                  masterDiskSize: Int, //min 10
+                                  // worker settings are None when numberOfWorkers is 0
+                                  workerMachineType: Option[String] = None,
+                                  workerDiskSize: Option[Int] = None, //min 10
+                                  numberOfWorkerLocalSSDs: Option[Int] = None, //min 0 max 8
+                                  numberOfPreemptibleWorkers: Option[Int] = None) extends RuntimeConfig {
+    val cloudService: CloudService = CloudService.Dataproc
+    val machineType: MachineType = MachineType(masterMachineType)
+    val diskSize: Int = masterDiskSize
+  }
+}
+
+// Information about service accounts used by the cluster
+final case class ServiceAccountInfo(clusterServiceAccount: Option[WorkbenchEmail],
+                                    notebookServiceAccount: Option[WorkbenchEmail])
+
+final case class ClusterError(errorMessage: String, errorCode: Int, timestamp: Instant)
```

Comment:
should this be RuntimeError?

Reply:
There's a clusterError table. So for refactoring this, it'll be a whole db refactoring...There're definitely more terms can be refactored. But I think this PR is mostly for refactoring out machineConfig in Cluster table